airbyte-integrations/connectors/source-monday/source_monday/graphql_request_options_provider.py

```diff
@@ -0,0 +1,111 @@
+import json
+import os
+from dataclasses import dataclass
+from typing import Optional, Mapping, Any, Union, MutableMapping, Type
+
+from airbyte_cdk.sources.declarative.interpolation import InterpolatedString
+from airbyte_cdk.sources.declarative.requesters.request_options import InterpolatedRequestOptionsProvider
+from airbyte_cdk.sources.declarative.types import StreamState, StreamSlice
+
+
+@dataclass
+class GraphQLRequestOptionsProvider(InterpolatedRequestOptionsProvider):
+    NEXT_PAGE_TOKEN_FIELD_NAME = "next_page_token"
+    NESTED_OBJECTS_LIMIT_MAX_VALUE = 100
+
+    limit: Union[InterpolatedString, str, int] = None
+
+    def __post_init__(self, options: Mapping[str, Any]):
+        super(GraphQLRequestOptionsProvider, self).__post_init__(options)
+
+        self.limit = InterpolatedString.create(self.limit, options=options)
+        self.name = options.get('name', '').lower()
+
+    def _ensure_type(self, t: Type, o: Any):
+        """
+        Ensure given object `o` is of type `t`
+        """
+        if not isinstance(o, t):
+            raise TypeError(f"{type(o)} {o} is not of type {t}")
+
+    def _get_schema_root_properties(self):
+        schema_path = os.path.join(os.path.abspath(os.curdir), "source_monday", f"schemas/{self.name}.json")
+        with open(schema_path) as f:
+            schema_dict = json.load(f)
+            return schema_dict["properties"]
+
+    def _get_object_arguments(self, **object_arguments) -> str:
+        return ",".join([f"{argument}:{value}" for argument, value in object_arguments.items() if value is not None])
+
+    def _build_query(self, object_name: str, field_schema: dict, **object_arguments) -> str:
+        """
+        Recursive function that builds a GraphQL query string by traversing given stream schema properties.
+        Attributes
+            object_name (str): the name of root object
+            field_schema (dict): configured catalog schema for current stream
+            object_arguments (dict): arguments such as limit, page, ids, ... etc to be passed for given object
+        """
+        fields = []
+        for field, nested_schema in field_schema.items():
+            nested_fields = nested_schema.get(
+                "properties",
+                nested_schema.get("items", {}).get("properties")
+            )
+            if nested_fields:
+                # preconfigured_arguments = get properties from schema or any other source ...
+                # fields.append(self._build_query(field, nested_fields, **preconfigured_arguments))
```

Comment:
looks like a debug code that probably should be removed

Reply:
This is not debug code. This is how I want to extend it but can`t for now

Reply:
can you add a comment explaining how you would want to use the commented code? Otherwise I think we should remove it if there's no context